zcash_client_sqlite/src/wallet.rs

```diff
@@ -986,6 +1036,49 @@ pub(crate) fn block_height_extrema(
     })
 }
 
+pub(crate) fn get_account<P: Parameters>(
+    conn: &rusqlite::Connection,
+    params: &P,
+    account_id: AccountId,
+) -> Result<Option<Account>, SqliteClientError> {
+    let mut sql = conn.prepare_cached(
+        r#"
+        SELECT account_type, uvk, hd_seed_fingerprint, hd_account_index
+        FROM accounts
+        WHERE id = :account_id
+    "#,
+    )?;
+
+    let mut result = sql.query(params![account_id.0])?;
+    let row = result.next()?;
```

Comment:
Why does this not use `sql.query_row(..).optional()`?

Reply:
Probably because I'm still learning rust and the rusqlite API. In trying to apply your suggestion, I'm hitting an issue that `query_row` really wants the lambda to return a rusqlite::Error type, yet we're returning `SqliteClientError` here.

Reply:
Mmk. In that case open an issue for fixing this and document it as a TODO inline.

Reply:
Is it broken as-is? What would the fix be?

Reply:
It won't be broken assuming that `AccountId` is unique (which is currently the case given that the `id` column is a `PRIMARY KEY`, but we can't guarantee indefinitely across database migrations). It's not clear from the type that `result` only contains one row, so it's preferable to use `query_row`, but it's not a blocker here.

Reply:
As to what the fix is, IDK, I'd have to spend more time playing around with the `rusqlite` API to figure out why the `Into` impls aren't working here. I don't have time to do that, so not a blocker.

Reply:
If `id` ceases to have a unique guarantee, that'll break a _lot_ more than just this migration. All APIs and queries are based on the expectation that it's unique.